src/server.h

```diff
@@ -579,7 +579,7 @@ typedef void *(*moduleTypeCopyFunc)(struct redisObject *fromkey, struct redisObj
 typedef int (*moduleTypeDefragFunc)(struct RedisModuleDefragCtx *ctx, struct redisObject *key, void **value);
 typedef void (*RedisModuleInfoFunc)(struct RedisModuleInfoCtx *ctx, int for_crash_report);
 typedef void (*RedisModuleDefragFunc)(struct RedisModuleDefragCtx *ctx);
-typedef size_t (*moduleTypeMemUsageFunc2)(struct RedisModuleKeyOptCtx *ctx, const void *value);
+typedef size_t (*moduleTypeMemUsageFunc2)(struct RedisModuleKeyOptCtx *ctx, const void *value, size_t sample_size);
```

Comment:
i think you're missing some update to the documentation of the `mem_usage2` field in module.c.
we can probably refer to the documentation of the MEMORY USAGE command, and i think we need to state that a value of 0 means infinite.

Reply:
`memoryCommand` already sets samples num to inf in case of 0, so sample_size arg in moduleTypeMemUsageFunc2 will always be >0.

Reply:
ok, not documenting it can't cause any harm, worse case, someone will implement code that handles the value of 0, and i'll be dead code.
so the only damage is some confusion for the module author.